### PR TITLE
EIP1-5615 / EIP1-5658 - Added mappings for electoralRollNumber, userId, deliveryAddressType

### PR DESCRIPTION
**Pull request by NathanRussellValtech**
This PR adds in the mappings for `electoralRollNumber`, `userId` and `deliveryAddressType` when re-issuing an AED
(because I forgot about them in my earlier PR! 🙄 )

### CODE REVIEW

File: src/main/kotlin/uk/gov/dluhc/printapi/mapper/aed/ReIssueAnonymousElectorDocumentMapper.kt [hunk 1]
```diff
@@ -35,13 +40,25 @@ abstract class ReIssueAnonymousElectorDocumentMapper {
...
     @Mapping(target = "delivery.address.id", ignore = true)
     @Mapping(target = "contactDetails.id", ignore = true)
     @Mapping(target = "contactDetails.address.id", ignore = true)
     abstract fun toNewAnonymousElectorDocument(
         previousAed: AnonymousElectorDocument,
+        dto: ReIssueAnonymousElectorDocumentDto,
         aedTemplateFilename: String
     ): AnonymousElectorDocument
+
+    @AfterMapping
```
Comment (NathanRussellValtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/mapper/aed/ReIssueAnonymousElectorDocumentMapper.kt:57]: Don't beat me up for this! It's the only way I could get it to work!

If I put `@Mapping(target = "delivery.deliveryAddressType", source = "dto.deliveryAddressType")` on the main mapping method, then the resultant `Delivery` object had all its fields null except `deliveryAddressType` !!!
It seems you have to let MapStruct map it from the previous AED, then mutate it afterwards! ¯\_(ツ)_/¯

**Review by vishalgupta-valtech — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/dluhc/printapi/mapper/aed/ReIssueAnonymousElectorDocumentMapperTest.kt [hunk 2]
```diff
@@ -101,15 +107,29 @@ class ReIssueAnonymousElectorDocumentMapperTest {
...
+
+        given(deliveryAddressTypeMapper.mapDtoToEntity(any())).willReturn(DeliveryAddressType.ERO_COLLECTION)
+
         val expected = previousAed.deepCopy().apply {
             certificateNumber = newCertificateNumber
             requestDateTime = FIXED_TIME
             issueDate = FIXED_DATE
             statusHistory = expectedStatusHistory.toMutableList()
+            electoralRollNumber = newElectoralRollNumber
+            delivery?.deliveryAddressType = DeliveryAddressType.ERO_COLLECTION
```
Comment (matt-wills-valtech) [inline src/test/kotlin/uk/gov/dluhc/printapi/mapper/aed/ReIssueAnonymousElectorDocumentMapperTest.kt:127]: How about checking that `delivery?.collectionReason` is also copied (which would be apt given the `DeliveryAddressType` here)?

Comment (vishalgupta-valtech) [reply src/test/kotlin/uk/gov/dluhc/printapi/mapper/aed/ReIssueAnonymousElectorDocumentMapperTest.kt:127]: I had similar question, but then I checked the code and found out that the mapper was copying entire delivery from previousAed, so looks good.

Comment (NathanRussellValtech) [reply src/test/kotlin/uk/gov/dluhc/printapi/mapper/aed/ReIssueAnonymousElectorDocumentMapperTest.kt:127]: OK , will do .... 👍

**Pull request merged**